### PR TITLE
update to get_secrets_key

### PR DESCRIPTION
**Pull request by MDunitz**
## Need
- to pass in the db_uri from AWS secrets manager to the default_dataset_config
- to raise an exception within get_secret_key and call sys.exit in the eb app when that exception is raised. 

## Approach
- split out server_attrs and default_dataset_attrs
- create error and move sys.exit back into eb/app.py

## Testing
- update unit test to include db_uri message

### CODE REVIEW

File: server/common/aws_secret_utils.py [hunk 1]
```diff
@@ -49,6 +54,18 @@ def handle_config_from_secret(app_config):
...
+    # update default dataset configuration attributes
+    for key, attr in default_dataset_attrs:
+        cur_val = getattr(app_config.default_dataset_config, attr)
+        if cur_val:
+            continue
+
+        # replace the attr with the secret if it is not set
+        val = secrets.get(key)
+        if val:
+            logging.error(f"set {attr} from secret")
```
Comment (bmccandless) [inline server/common/aws_secret_utils.py:66]: I'm not sure why I had this as logging.error.  It should be "info".  It is definitely not an error.

File: server/test/unit/common/test_app_config.py [hunk 2]
```diff
@@ -121,10 +122,11 @@ def test_get_config_vars_from_aws_secrets(self, mock_get_secret_key):
...
             from server.common.aws_secret_utils import handle_config_from_secret
             # should not throw error
-            # "AttributeError: 'ServerConfig' object has no attribute 'user_annotations__hosted_tiledb_array__db_uri'"
+            # "AttributeError: 'XConfig' object has no attribute 'x'"
             handle_config_from_secret(config)
 
             # should throw 2 errors (one for each var set from a secret)
-            self.assertEqual(len(logger.output), 2)
+            self.assertEqual(len(logger.output), 3)
             self.assertIn('ERROR:root:set app__flask_secret_key from secret', logger.output[0])
```
Comment (bmccandless) [inline server/test/unit/common/test_app_config.py:130]: You can also directly check for success by checking the config, e.g:
```
self.assertEqual(confg.server_config.app__flask_secret_key, "mock_flask_secret")
```

File: server/common/aws_secret_utils.py (hunk 1 above)
Comment (MDunitz) [reply server/common/aws_secret_utils.py:66]: ok cool, will update

File: server/test/unit/common/test_app_config.py (hunk 2 above)
Comment (MDunitz) [reply server/test/unit/common/test_app_config.py:130]: oh good call!

**Pull request merged**